File: Homeworks/homework1/torun_boran.php
<?php
// Function definition of foo in php
function foo($key, $value)
{
    echo $key . ":" . $value;
    echo "\n";
}

// Initial array
echo "Initial array";
echo "\n";
$assocArr = array("School" => "Bilkent", "City" => "Ankara", "Year" => 1984);
foreach ($assocArr as $key => $value)
{
    foo($key, $value);
}
echo "\r\n";

// Get the value for a given key
echo "Get the value for a given key";
echo "\r\n";
echo "School is ". $assocArr["School"];
echo "\r\n";
echo "\r\n";

// Add a new element
echo "Add a new element";
echo "\n";
$assocArr += array("District" => "Çankaya");
foreach ($assocArr as $key => $value)
{
    foo($key, $value);
}
echo "\r\n";

// Remove an element
echo "Remove an element";
echo "\n";
unset($assocArr["Year"]);
foreach ($assocArr as $key => $value)
{
    foo($key, $value);
}
echo "\r\n";

// Modify the value of existing element
echo "Modify the value of existing element";
echo "\r\n";
$assocArr["School"] = "METU";
foreach ($assocArr as $key => $value)
{
    foo($key, $value);
}
echo "\r\n";

// Search for the existence of a key
echo "Search for the existence of a key";
echo "\r\n";
if(array_key_exists("Founder", $assocArr))
{
    echo "This associative array contains this key";
    echo "\r\n";
}
else
{
    echo "This associative array doesn't contain this key";
    echo "\r\n";
}
echo "\r\n";

// Search for the existence of a value
echo "Search for the existence of a value";
echo "\r\n";
if(in_array("METU", $assocArr))
{
    echo "This associative array contains this value";
    echo "\r\n";
}
else
{
    echo "This associative array doesn't contain this value";
    echo "\r\n";
}

echo "\r\n";

echo "Print the key-value pairs";
echo "\r\n";
foreach ($assocArr as $key => $value)
{
    foo($key, $value);
}

?>

File: Homeworks/homework1/torun_boran.rs
use std::collections::HashMap;

// Function definition of foo in Rust
fn foo(key: &str, value: &str)
{
    println!("({} : {})", key, value);
}

fn main() {
    // Initial array
    println!("Initial array");
    let mut assoc_array = HashMap::from([("cloth type", "t-shirt"), ("size", "XL")]);
    for (k, v) in &assoc_array
    {
        foo(k, v);
    }
    
    println!();
    
    // Get the value for a given key
    println!("Get the value for a given key");
    print!("Cloth type is ");
    println!("{}", assoc_array["cloth type"]);
    println!();
    
    // Add a new element
    println!("Add a new element");
    assoc_array.insert("brand", "Network");
    for (k, v) in &assoc_array
    {
        foo(k, v);
    }
    println!();
    
    // Remove an element
    println!("Remove an element");
    assoc_array.remove("size");
    for (k, v) in &assoc_array
    {
        foo(k, v);
    }
    println!();
    
    // Modify the value of an existing element
    println!("Modify the value of an existing element");
    assoc_array.insert("cloth type", "jacket");
    println!();
    
    // Search for the existence of a key
    println!("Search for the existence of a key");
    if assoc_array.contains_key("brand")
    {
        println!("This associative array contains this key");
    }
    else
    {
        println!("This associative array doesn't contain this key");
    }
    println!();
    
    // Search for the existence of a value
    println!("Search for the existence of a value");
    let mut i = 0;
    for v in assoc_array.values()
    {
        if v == &"jacket"
        {
            i = 1;
            println!("This associative array contains this value");
        }
    }
    if i != 1
    {
        println!("This associative array doesn't contain this value");
    }
}